src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,15 @@
+import React, { useContext } from 'react';
+import { TodosContext } from '../TodosContext';
+import { TodoItem } from '../TodoItem';
+
+export const TodoList: React.FC = () => {
+  const { visibleTodos } = useContext(TodosContext);
+
+  return (
+    <ul className="todo-list">
+      {visibleTodos.map(todo => (
+        <TodoItem todo={todo} />
```

Comment:
you can't separate ul and li tags, put li here, also add key in this component
```
<ul className="todo-list">
    {visibleTodos.map(todo => (
      <li key={todo.id}>
        <TodoItem todo={todo} />
```

Reply:
return (
    <ul className="todo-list">
      {visibleTodos.map(todo => (
        <TodoItem key={todo.id} todo={todo} />
      ))}
    </ul>